src/explore-education-statistics-common/src/components/form/FormCheckboxGroup.tsx

```diff
@@ -40,45 +41,46 @@ export type CheckboxGroupAllChangeEventHandler = (
 
 interface BaseFormCheckboxGroupProps {
   disabled?: boolean;
+  groupLabel: string;
   id: string;
   name: string;
   options: CheckboxOption[];
-  selectAll?: boolean;
-  selectAllText?: (isAllChecked: boolean, options: CheckboxOption[]) => string;
-  small?: boolean;
   order?: OrderKeys<CheckboxOption>;
   orderDirection?: OrderDirection | OrderDirection[];
+  selectAll?: boolean;
+  selectAllText?: (
+    isAllChecked: boolean,
+    options: CheckboxOption[],
+  ) => string | ReactNode;
+  small?: boolean;
   value: string[];
   onAllChange?: CheckboxGroupAllChangeEventHandler;
   onBlur?: FocusEventHandler<HTMLInputElement>;
   onChange?: CheckboxChangeEventHandler;
-  hiddenText?: string;
   inputRef?: Ref<HTMLInputElement>;
 }
 
-const getDefaultSelectAllText = (
-  isAllChecked: boolean,
-  opts: CheckboxOption[],
-) => `${isAllChecked ? 'Unselect' : 'Select'} all ${opts.length} options`;
+export const getDefaultSelectAllText = (isAllChecked: boolean, total: number) =>
+  `${isAllChecked ? 'Unselect' : 'Select'} all ${total} options`;
```

Comment:
Should we add a check here to not pluralize `options` if the total is only one?

Reply:
I'll check, but I don't think you get the select all button if there's only one option.

Reply:
I've checked now and the select all button will only show if there's more than one option so this is ok always being plural.

Reply:
I don't think it's necessary as this is specifically text for selecting all of a list of options. If you were using this where there was only one option you'd have a button that says "Select option" above a single checkbox which would be odd and potentially confusing.

Reply:
I'd personally fix it because there's no real downside to coding defensively here. It's feasible that this function gets used incorrectly in the future so we could end up having a button that says 'Select/Unselect all 1 options' despite any best intentions here.

Anyway, if you feel particularly strongly about it, I'll approve the PR anyway, but I would suggest just making the change.

Reply:
I just feel like defensive coding here would make it more likely that we'd use it wrongly as it would seem like you're meant to show the button when there's only one option. So I'll leave it in for now.